### ISSUE TITLE
Does the SDK respect DNS TTL?

### ISSUE THREAD

**Issue by rclark** [aws/aws-xray-sdk-go#263]
```
2020-11-10T16:14:13Z [DEBUG] Error occurred while refreshing sampling rules. RequestError: send request failed
caused by: Post "http://172.31.133.144:443/GetSamplingRules": dial tcp 172.31.133.144:443: i/o timeout
```

I have a container that's been throwing this error for the last hour or so. `172.31.133.144` is a stale IP address for the DNS name of my daemon service, which was used in the `xray.Configure()` step when the container launched. That DNS record has a low TTL (60s). Does the SDK respect and re-resolve DNS when TTL periods expire?

**Comment by bhautikpip** [comment]
Hi @rclark ,

Do you think SDK is using `172.31.133.144` to send data to X-Ray Daemon? I don't think SDK respects DNS TTL in the scenario you mentioned. Why this ip `172.31.133.144` becomes stale? Would you be able to explain how you're using X-Ray SDK with X-Ray daemon setup?

**Comment by rclark** [comment]
I'm running the xray daemon on a set of ECS tasks, using AWS CloudMap as the discovery layer. Those ECS containers can stop for any number of reasons. It looks like the daemon resolves DNS once on bootstrap, but doesn't refresh after the DNS-defined TTL.

**Comment by bhautikpip** [comment]
Yes, @rclark I think the part of the reason you're seeing such logs could be because when new ECS containers spins up SDK still used the stale IP address that you had. As far as I know SDK doesn't support anything like this. This can be added to SDK as an enhancement. I will add this task in our backlog but meanwhile we do appreciate PRs :)

**Comment by rclark** [comment]
Thanks for the info.  

If my application runs `xray.Configure()` more than once, would that re-resolve the DNS name to a new IP address each time I run it?
